### ISSUE TITLE
About training speed and dataset.

### ISSUE THREAD

**Issue by YamamotoSayaka** [rosinality/style-based-gan-pytorch#73]
Hi, I'm quite new to GAN, so this might look like a stupid question.
I'm trying to rerun your code on the celeba dataset on one single TitanX GPU.

I only used the img_align_celeba.zip(1.4G) from the official website, extractd it, and converted it to the lmdb format.
And I looked into the training log, it would take me around 371 hours to finish it. Is it normal? Or am I doing something wrong? Or maybe should I use the celebaHD following the original paper?

0%|                                                 | 43/3000000 [00:21<371:05:56,  2.25it/s]

Many thanks!

**Comment by kevinstan** [comment]
Hi @YamamotoSayaka , I think it is normal. To my understanding, the max iter is set arbitrary high to 3000000 probably to allow the model to train until convergence. 

In my experience, I'm training on FFHD on 8 TITAN X GPU, and by the time the model reaches 512x512 resolution, the estimated finish time is extremely long (2000+ hrs)

Size: 512; G: 1.216; D: 1.170; Grad: 0.060; Alpha: 0.84739:   5%|‚ñå         | 161250/3000000 [55:52:40<2025:07:15,  2.57s/it]
Size: 512; G: 1.216; D: 1.170; Grad: 0.060; Alpha: 0.84739:   5%|‚ñå         | 161251/3000000 [55:52:40<2026:02:55,  2.57s/it]
Size: 512; G: 1.216; D: 1.170; Grad: 0.060; Alpha: 0.84744:   5%|‚ñå         | 161251/3000000 [55:52:43<2026:02:55,  2.57s/it]
Size: 512; G: 1.216; D: 1.170; Grad: 0.060; Alpha: 0.84744:   5%|‚ñå         | 161252/3000000 [55:52:43<2026:14:02,  2.57s/it]

As a reference, the original paper stated that they trained for approx one week on 8 Tesla V100. For CelebA I suspect the training to be much faster since it is a smaller dataset compared to FFHD (although need clarification). 

Another idea is to plot the train loss to visualize when the model converges and then stop the training early in a non-rigorous way when you're happy with the quality of the samples. 

@rosinality Pls confirm if this seems normal

**Comment by rosinality** [comment]
@YamamotoSayaka @kevinstan Yes, it is set to be arbitrary high. If we follow papers 25M settings, max iterations will be about 700k, as:

Resolution | # Images | Batch size | # iters
---------------|-------------|----------------|------------
8 | 120k | 256 | 4,688
16 | 120k | 128 | 9,375
32 | 120k | 64 | 18,750
64 | 120k | 32 | 37,500
128 | 120k | 32 | 37,500
256 | 120k | 32 | 37,500
512 | 120k | 32 | 37,500
1024 | 25M - 120k * 7 = 520k | 32 | 520k

But train iters will be further slow down as resolution increases. So I think it can require a lot of time if your target resolution is large.

**Comment by YamamotoSayaka** [comment]
@kevinstan ,@rosinality Thanks very much for your quick reply!
Let's say if I just want a quick result from the code, for example, I only run on the first 5k images from celeba. As long as I reach my desired resolution, I can just stop when I have a satisfactory result, right?

**Comment by rosinality** [comment]
@YamamotoSayaka Yes, you can stop at anytime.

**Comment by YamamotoSayaka** [comment]
@rosinality Many thanks!!
